src/ctia/flows/hooks/event_hook.clj

```diff
@@ -0,0 +1,23 @@
+(ns ctia.flows.hooks.event-hook
+  (:require
+   [ctia.flows.hook-protocol :refer [Hook]]
+   [ctia.events.producers.es.producer :as esp]))
+
+(defrecord ESEventProducerRecord [conn]
+  Hook
+  (init [_] (do
+              (reset! conn (esp/init-producer-conn))
+              (comment println "ESEventProducerRecord Initialized: " (pr-str @conn))))
```

Comment:
Ah yes, I didn't made it really clear. But `init` and `destroy` functions should contains list of function for which the return value as not any importance. These are just function to give a chance to have some stateful effect, typically, asking for resources.

I should write better comment for the `Hook` protocol.